src/components/GrantsOfficeHome.vue

```diff
@@ -46,6 +61,13 @@ export default {
   components: {
     UploadHistory
   },
+  mounted () {
+    window.addEventListener('focus', this.clearLoadingState.bind(this))
```

Comment:
huh, how did you know that you need to re-bind `this`?

Reply:
Ooph, JavaScript has some real rough edges when working with the `this` keyword.  I would say the reliance on `this` is my least favourite part of the Vue framework.

I don't even know where to start with an explanation!

I guess the main concept is that `this` in a JS function is generally determined by the context in which the function is _called_ and not the context in which it is defined.  That means that if a function has references to `this` and you intend to pass it as a callback somewhere else, then you can assume it those `this` references will be transferred to the calling code — unless you bind the current this value before passing.

This particular pattern shows up a lot in Vue (and in the class-style React components).  If you have a component method or lifecycle hook (which probably refers to this), but you plan to pass it as a callback somewhere, you need to rebind `this`, or it won't do what you expected.

There are some notes on the pattern in React docs, here:

https://reactjs.org/docs/handling-events.html

---

With all that said, I wrote it this way expecting it to be necessary, but I didn't actually test without.

I just poked at the VueJS docs, and at least in the latest version, it claims that the framework binds all functions in the `methods` object automatically, so it might not be required here!

https://github.com/vuejs/vue/blob/33e669b/src/core/instance/state.js#L286